week-2-wide-perspective/w02e05-simple-server-side/src/middlewares/auth.ts

```diff
@@ -0,0 +1,31 @@
+import { NextFunction, Request, Response } from "express";
+
+import { Token, verifyAccessToken } from "../utils/jwt";
+import { prisma } from "../../prisma/db";
+
+export const auth = async (req: Request, res: Response, next: NextFunction) => {
+  if (!req.headers.authorization) {
+    return res.status(403).send("A token is required for authentication");
```

Comment:
tutaj odpowiadasz jako:
```
Content-Type: text/html; charset=utf-8
```

a to się nie pokrywa z `.json` którego używasz w innych miejscach - jak zwracasz dane.

Powinien być jeden rodzaj odpowiedzi z serwera.
Wszystko leci jako `application/json`

Możesz przygotować sobie własną klasę błędu + globalErrorHandler:

https://expressjs.com/en/guide/error-handling.html#the-default-error-handler

I tutaj po prostu rzucać błędem + status - a tam łapać i wysyłać jako sformatowany `json` + info.